webapp/packages/plugin-data-viewer/src/ValuePanelPresentation/TextValue/useTextValue.ts

```diff
@@ -30,63 +31,60 @@ interface IUseTextValue {
   pasteFullText(): Promise<void>;
 }
 
-export function useTextValue({ model, resultIndex, currentContentType }: IUseTextValueArgs): IUseTextValue {
-  const { formatAction, editAction, contentAction, dataAction } = useResultActions({ model, resultIndex });
-  const selection = model.source.getAction(resultIndex, ResultSetSelectAction);
-  const activeElements = selection.getActiveElements();
-  const firstSelectedCell = activeElements?.[0];
+export function useTextValue({ model, resultIndex, currentContentType, elementKey }: IUseTextValueArgs): IUseTextValue {
+  const { formatAction, editAction, contentAction } = useResultActions({ model, resultIndex });
   const formatter = useAutoFormat();
-  const columnType = firstSelectedCell ? dataAction.getColumn(firstSelectedCell.column)?.dataKind : '';
-  const isTextColumn = columnType?.toLocaleLowerCase() === 'string';
-  const contentValue = firstSelectedCell ? formatAction.get(firstSelectedCell) : null;
-  const cachedFullText = firstSelectedCell ? contentAction.retrieveFileFullTextFromCache(firstSelectedCell) : '';
-  const blob = firstSelectedCell ? formatAction.get(firstSelectedCell) : null;
+  const isTextColumn = elementKey ? formatAction.isText(elementKey) : false;
+  const contentValue = elementKey ? formatAction.get(elementKey) : null;
+  const isBinary = elementKey ? formatAction.isBinary(elementKey) : false;
+  const cachedFullText = elementKey ? contentAction.retrieveFileFullTextFromCache(elementKey) : '';
   const notificationService = useService(NotificationService);
+  const limitInfo = useResultSetValueLimitInfo({ model, resultIndex, elementKey });
 
   const result: IUseTextValue = {
     textValue: '',
     isTruncated: false,
     isTextColumn,
     async pasteFullText() {
-      if (!firstSelectedCell) {
+      if (!elementKey) {
         return;
       }
 
       try {
-        await contentAction.getFileFullText(firstSelectedCell);
+        await contentAction.getFileFullText(elementKey);
       } catch (exception) {
         notificationService.logException(exception as any, 'data_viewer_presentation_value_content_paste_error');
       }
     },
   };
 
-  if (!isNotNullDefined(firstSelectedCell)) {
+  if (!isNotNullDefined(elementKey)) {
     return result;
   }
 
-  if (isResultSetContentValue(contentValue)) {
-    result.isTruncated = contentAction.isContentTruncated(contentValue);
+  if (isResultSetContentValue(contentValue) && limitInfo.limit) {
+    result.isTruncated = (contentValue?.contentLength ?? 0) > limitInfo.limit;
```

Comment:
`ResultSetDataContentAction.isContentTruncated` have wrong implementation that works only for binary data seems like logic from `useResultSetValueLimitInfo` can be moved to `isContentTruncated` function